Methods_and_interfaces/18Stringers.go

```diff
@@ -0,0 +1,30 @@
+package main
+
+import "fmt"
+
+/*
+ * 学習内容
+ * ・インターフェイスはメソッドを複数記述し、リスト化する。
+ * ・メソッドはレシーバ引数を取る関数。typeで型定義されたものをレシーバ引数として取る。
+ */
+
+// byte型で4個の変数が格納できる配列IPAddrを作成
+type IPAddr [4]byte
+
+// IPAddrにインターフェイスfmt.Stringを実装する。
+// fmt.Sprintfと%dを使用することで配列aをstring型に変更して返り値とする
+func (a IPAddr) String() string {
+	return fmt.Sprintf("%d.%d.%d.%d", a[0], a[1], a[2], a[3])
+}
+
+func main() {
+	// mapの作成 キーはstring型、値はIPAddrで処理を行ったstring型とする
+	hosts := map[string]IPAddr{
+		"loopback":  {12, 0, 0, 1},
+		"googleDNS": {8, 8, 8, 8},
+	}
+	// 繰り返し処理を実施。キーをname、値をipとして出力する
+	for name, ip := range hosts {
+		fmt.Printf("%v: %v\n", name, ip)
```

Comment:
こうかなと思います。
fmtパッケージで呼び出すことにより、fmt.String interfaceを通って↑のIPAddr.String()が発火します。
```suggestion
		fmt.Println(ip)
```

Reply:
ここだけわからなかったです。
fmt.Println(ip)とすると、nameが使われていないのでエラーが発生するのと、fmt.PrintfをPrintlnに直す理由もわからなかったです。明日また質問します！

Reply:
> nameが使われていないのでエラー

使われない変数を定義すると怒られます。
ですが、rangeの返り値は2つあるので受け取らざるを得ない。
そんなときは、`name`を`_`にして無視しましょう。

> fmt.PrintfをPrintlnに直す理由
Printfはフォーマット指定してparamを文字列出力するもので、
Printlnは、ただ文字列出力するだけ（と改行か）。
String()メソッドで　整形したstringが返ってくるのでフォーマット指定する必要はないということですね。

Reply:
理解しました。確かに整形したものをさらにフォーマット指定する必要はないですね。
ありがとうございます！